From: Michael Biebl <biebl@debian.org>
Date: Sun, 29 Mar 2015 22:58:43 +0200
Subject: Don't block network.target on NetworkManager-wait-online.service

network.target is supposed to be a passive unit which is pulled in my
the network management service.

Also, we want NetworkManager-wait-online.service be hooked up in
network-online.target, not multi-user.target.

While at it, add a few other smaller fixes, like the addition of
Documentation= and RemainAfterExit=yes

http://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/
https://bugzilla.gnome.org/show_bug.cgi?id=746039
---
 data/Makefile.am                           | 4 ----
 data/NetworkManager-wait-online.service.in | 7 ++++---
 2 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/data/Makefile.am b/data/Makefile.am
index a7c8070..60443bf 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -18,10 +18,6 @@ endif
 
 NetworkManager-dispatcher.service: NetworkManager-dispatcher.service.in
 	$(edit) $< >$@
-
-install-exec-local:
-	install -d $(DESTDIR)$(systemdsystemunitdir)/network-online.target.wants
-	ln -sf $(systemdsystemunitdir)/NetworkManager-wait-online.service $(DESTDIR)$(systemdsystemunitdir)/network-online.target.wants
 endif
 
 examplesdir = $(docdir)/examples
diff --git a/data/NetworkManager-wait-online.service.in b/data/NetworkManager-wait-online.service.in
index 6ad6942..1753d20 100644
--- a/data/NetworkManager-wait-online.service.in
+++ b/data/NetworkManager-wait-online.service.in
@@ -1,13 +1,14 @@
 [Unit]
 Description=Network Manager Wait Online
+Documentation=man:nm-online(1)
 Requisite=NetworkManager.service
 After=NetworkManager.service
-Wants=network.target
-Before=network.target network-online.target
+Before=network-online.target
 
 [Service]
 Type=oneshot
 ExecStart=@bindir@/nm-online -s -q --timeout=30
+RemainAfterExit=yes
 
 [Install]
-WantedBy=multi-user.target
+WantedBy=network-online.target
